src/test/java/org/mutabilitydetector/locations/DottedTest.java

```diff
@@ -0,0 +1,164 @@
+package org.mutabilitydetector.locations;
+
+/*-
+ * #%L
+ * MutabilityDetector
+ * %%
+ * Copyright (C) 2008 - 2019 Graham Allan
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import static org.junit.Assert.assertEquals;
+import static org.mutabilitydetector.locations.CodeLocation.FieldLocation.fieldLocation;
+
+import java.lang.reflect.Field;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.junit.Test;
+import org.mutabilitydetector.TestUtil;
+import org.mutabilitydetector.checkers.AsmMutabilityChecker;
+import org.mutabilitydetector.locations.CodeLocation.ClassLocation;
+import org.mutabilitydetector.locations.CodeLocation.FieldLocation;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+
+public class DottedTest {
```

Comment:
Seems a bit strange to call this class `DottedTest`. It's more of a `FieldLocation` or maybe even an `AsmMutabilityChecker` test. 

:+1: for writing a test that compares ASM vs. Java reflection, nice choice.

Reply:
The method under test is really `org.mutabilitydetector.locations.Dotted.fromFieldDescription(String)`, which is why I ended up putting it here. I call it on line 54. I extended `AsmMutabilityChecker` for the test because I had to make sure I was getting the format of the `desc` value correctly for all possible use cases. That's how I found arrays had to be handled differently. But I'm happy to put the tests elsewhere if that helps maintainability.

Btw., I did struggle a bit on where to put the description parsing logic. At first, I thought it could be a protected method on `AsmMutabilityChecker`, but I found scenarios where it had to be handled that were outside of that class hierarchy.

So let me know if you'd like me to move these tests and if so where. Also let me know if you feel the parsing logic should be placed elsewhere.

Reply:
You've convinced me about the unit-under-test as well as where the method should live. Changed my mind, consider this a :thumbsup: